device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttTransport.java

```diff
@@ -343,4 +343,15 @@ public boolean isEmpty()
 
         return false;
     }
+
+    /**
+     * Registers a callback to be executed whenever the mqtt connection is lost or established.
+     * 
+     * @param callback the callback to be called.
+     */
+    public void registerConnectionStateCallback(IotHubConnectionStateCallback callback) {
+        //TODO: Add in registration logic if needed.  The transport would also need to be updated to
+        //      receive notifications from the MQTT Connection class when the connection is lost or
+        //      established.
```

Comment:
If you wish to leave this as is then please throw UnSupportedOperation here. If not please implement the logic.